### DISCUSSION TITLE
Plugin to support R Markdown syntax for code chunks/fenced code blocks

### DISCUSSION THREAD

**Discussion by rgaiacs** [remarkjs/remark#638] [Ideas] [answered]
I use [R Markdown](https://rmarkdown.rstudio.com/) a lot and I'm interested to use remark and retext for some personal projects related with R Markdown. As explained at https://github.com/rstudio/rmarkdown/issues/1073#issuecomment-306924809, R Markdown uses curly braces to indicate a code block that need to be executed/interpreted, for example

``````
```{r}
x = 1 + 1
```
``````

I would love to have a mentor to help me write a plugin like https://github.com/s0/remark-code-frontmatter. Thanks in advance.

**Comment by ChristianMurphy** [comment]
https://unifiedjs.com/learn gives a good introduction to remark/unified and writing plugins.
remark can parse the code block with curly around language name syntax described above https://astexplorer.net/#/gist/c9c1aa4d51de45d60f89e9a1a88289dd/27e1400acbd7ba507b66cdbb424164e17370e5b7
https://github.com/remarkjs/remark/blob/main/doc/plugins.md has a lot of examples of existing plugins that can be used for inspiration.

**Comment by rgaiacs** [comment, marked as answer]
Thanks for the message. I started https://github.com/rgaiacs/remark-rmarkdown with a minimal code based on https://unifiedjs.com/learn.

R Markdown has code blocks like

``````
```{r example, echo = FALSE}
x = 1 + 1
```
``````

which generate a AST like

```
      {
      "type": "code",
      "lang": "{r",
      "meta": "example, echo = FALSE}",
      "value": "x = 1 + 1",
      "position": {
        "start": {
          "line": 1,
          "column": 1,
          "offset": 0
        },
        "end": {
          "line": 3,
          "column": 4,
          "offset": 42
        }
      }
```

My plugin removes the `{` from `lang` and `}` from `meta`.
